syntax = "proto2";

package pb;

import "common.proto";

message E42mConfig{
	required uint32 N = 1;
	optional double delta_0 = 2;
	optional double alpha = 3;
}

message E42mState{
	required double E_p = 1;
	required double phi_p = 2;
	repeated group Particles = 3{
		required double a = 4;
		required double psi = 5;
		required double z = 6;
		required double delta = 7;
	}
	required double E_m = 8;
	required double phi_m = 9;	
}

message E42mStateGeneratorConfig{
	required double E_p0 = 1;
	required double phi_p0 = 2;
	required double A0 = 3;
	required double E_m0 = 4;
	required double phi_m0 = 5;	
}

message E42mModel{
	required E42mConfig pconfig = 1;
	required E42mState state = 2;
	required EXPetscSolverConfig sconfig = 3;
}

message E42mSolution{
	required E42mState state = 1;
	optional E42mState d_state = 2;
}

message E42mSpecial{
	optional double e_p_2 = 1;
	optional double e_m_2 = 16;
	optional double aver_a_2 = 2;
	optional double int_e_a = 4;
	optional double Na = 5;
	optional double Nb = 6;
	optional double M = 17;
	optional double N = 18;
	optional double Ia = 7;
	optional double Ib = 8;
	optional double Ia_aver = 9;
	optional double Ib_aver = 10;	
	optional double Wa = 11;
	optional double Wb = 12;
	repeated group Hist = 13 {
		required double x = 14;
		required double y = 15;
	}
}